wasmer/trap.go

```diff
@@ -24,12 +24,34 @@ type Trap struct {
 	_ownedBy interface{}
 }
 
-func newTrap(pointer *C.wasm_trap_t, ownedBy interface{}) *Trap {
+// newWasmTrap creates C wasm_trap structure and returns it's pointer
+func newWasmTrap(store *Store, message string) *C.wasm_trap_t {
+	messageBytes := []byte(message)
+	var bytesPointer *C.uint8_t
+	bytesLength := len(messageBytes)
+
+	if bytesLength > 0 {
+		bytesPointer = (*C.uint8_t)(unsafe.Pointer(&messageBytes[0]))
+	}
+
+	runtime.KeepAlive(store)
+	runtime.KeepAlive(message)
+
+	return C.to_wasm_trap_new(store.inner(), bytesPointer, C.size_t(bytesLength))
+}
+
+// newTrapFromPonter creates Trap from C.wasm_trap structure and attaches GC finalizer for it
```

Comment:
Small typo on the documented method name
```suggestion
// newTrapFromPointer creates Trap from C.wasm_trap structure and attaches GC finalizer for it
```